pkg/virt-controller/services/template.go

```diff
@@ -258,24 +260,23 @@ func (t *templateService) RenderLaunchManifestNoVm(vmi *v1.VirtualMachineInstanc
 	return t.renderLaunchManifest(vmi, nil, true)
 }
 
-func (t *templateService) RenderMigrationManifest(vmi *v1.VirtualMachineInstance, pod *k8sv1.Pod) (*k8sv1.Pod, error) {
-	imageIDs := containerdisk.ExtractImageIDsFromSourcePod(vmi, pod)
-	podManifest, err := t.renderLaunchManifest(vmi, imageIDs, false)
+func (t *templateService) RenderMigrationManifest(vmi *v1.VirtualMachineInstance, sourcePod *k8sv1.Pod) (*k8sv1.Pod, error) {
+	imageIDs := containerdisk.ExtractImageIDsFromSourcePod(vmi, sourcePod)
+	targetPod, err := t.renderLaunchManifest(vmi, imageIDs, false)
 	if err != nil {
 		return nil, err
 	}
 
-	if namescheme.PodHasOrdinalInterfaceName(network.NonDefaultMultusNetworksIndexedByIfaceName(pod)) {
-		ordinalNameScheme := namescheme.CreateOrdinalNetworkNameScheme(vmi.Spec.Networks)
-		multusNetworksAnnotation, err := multus.GenerateCNIAnnotationFromNameScheme(
-			vmi.Namespace, vmi.Spec.Domain.Devices.Interfaces, vmi.Spec.Networks, ordinalNameScheme, t.clusterConfig)
+	if t.netAnnotationsConverter != nil {
+		convertedNetAnnotations, err := t.netAnnotationsConverter.Convert(vmi, sourcePod)
```

Comment:
I do not understand what this converts exactly. To me it just creates the annotation based on information coming from the VMI and pod.

I also think it does not fit well with the pattern used for other dependency injections as it is too specific.
Why it is not part of the annotations generators interface? After all, it is also an annotation generation action, just with an additional input, fitting a specific location (preparing a migration target pod).

What do you say about one of these options?
- Add another method for generating the annotations based on another input (pod).
- Take this out of here and move it to the caller (the migration controller), having a hook there for network manipulation on the target pod.